assets/js/base/context/cart-checkout/shipping/index.js

```diff
@@ -119,28 +92,22 @@ export const ShippingDataProvider = ( { children } ) => {
 
 	// increment/decrement checkout calculating counts when shipping is loading
 	useEffect( () => {
-		if ( shippingOptionsLoading ) {
+		if ( shippingRatesLoading ) {
 			dispatchActions.incrementCalculating();
 		} else {
 			dispatchActions.decrementCalculating();
 		}
-	}, [ shippingOptionsLoading ] );
+	}, [ shippingRatesLoading ] );
 
-	// @todo need to add error handling to useShippingRates so that errors are
-	// exposed. We need error types exposed by the error handling as well.
-	// also we need to add similar logic for selection/unselection of rates and
-	// emit the events (see emit events block)
-	const onRateChange = ( shippingRates, shippingRatesLoading, error ) => {
-		setShippingOptions( shippingRates );
-		setShippingOptionsLoading( shippingRatesLoading );
-		if ( ! shippingRatesLoading && error && error.type ) {
-			// @todo this type might need normalizing to something recognized by
-			// the ERROR_TYPE constants.
-			dispatchErrorStatus( setStatusAction( error.type ) );
-		} else if ( ! shippingRatesLoading && shippingRates ) {
-			dispatchErrorStatus( NONE );
```

Comment:
Originally I thought we'd be potentially getting errors like invalid address/country type errors from the api. However, with the current error handling in `useShippingRates` and `useSelectShippingRate` hooks, the assumption is being made that any error coming from the api is a non-recoverable error because of the current ui/ux setup (known countries, states etc).

This is subject to being revisited at some point because it may still be possible for extensions to hook in eventually and they might provide invalid values (from a ui/ux we have less control over).  If needed we can update in future work (which will include handling appropriate error types in the relevant hooks).